doc/specs/#1620 - Installer selector on multiple matches.md

```diff
@@ -0,0 +1,235 @@
+---
+author: Darshan Rander @SirusCodes
+created on: 2022-10-29
+last updated: 2022-11-11
+issue id: 1620
+---
+
+# Spec Title
+
+For [#1620](https://github.com/microsoft/winget-cli/issues/1620)
+
+## Abstract
+
+Currently, the winget.exe client simply lists all the matches and if the user wants to install the app they need to re-run the command with Id.
+
+The UX can be improved here by giving users an option to select the package instead of asking them to re-run the command with Id.
+
+This can be extended to `upgrade`, `show` and `uninstall` commands.
+
+## Inspiration
+
+Whenever I use Winget to install any application there are high chances that there are name collisions with other names and I get multiple suggestions in the table and it's annoying to copy the ID and rerun the command. I guess this is the case for a lot of others as well.
+
+## Solution Design
+
+If we find multiple packages then the user should get a selector.
+
+> It should not be shown if `--diable-interactivity` is set.
+
+### Settings
+
+The user needs to **opt-in** to this feature.
+
+Winget can use `--prompt` and `--no-prompt` flags to enable and disable it respectively.
+
+The user can change the default setting to always show the prompt by -
+
+```pwsh
+"packageDisambiguation": "prompt"
+```
+
+and disable it by -
+
+```pwsh
+"packageDisambiguation": "no-prompt"
+```
+
+### Valid inputs
+
+- Number - It should match with a _row number_ (0 will exit the selector)
+- Text - It should match either the package _Name_ or _Id_
+- `--source` flag - It refines the search with a specific package source. It can only be used if a user enters text or standalone.
```

Comment:
I would extend this to be any of the default query flags. Essentially, I think of the table results as a new source that can be queried against, and whatever is input is the [[-q] <query>] for a search against the new source.

Probably don't need things like `-n` or `--header` or `--accept-source-agreements`, but I would imagine that a majority of the options would still be available
```
Searches for packages from configured sources.

usage: winget search [[-q] <query>] [<options>]

The following command aliases are available:
  find

The following arguments are available:
  -q,--query                  The query used to search for a package

The following options are available:
  --id                        Filter results by id
  --name                      Filter results by name
  --moniker                   Filter results by moniker
  --tag                       Filter results by tag
  --cmd,--command             Filter results by command
  -s,--source                 Find package using the specified source
  -n,--count                  Show no more than specified number of results (between 1 and 1000)
  -e,--exact                  Find package using exact match
  --header                    Optional Windows-Package-Manager REST source HTTP header
  --accept-source-agreements  Accept all source agreements during source operations
```

Reply:
I never thought it as CLI inside a CLI. Wouldn't that be an overkill for a simple thing? 👀

Reply:
> I never thought it as CLI inside a CLI. Wouldn't that be an overkill for a simple thing? 👀

It isn’t a CLI in a CLI though.

I'll use the example of the default `winget` source. That source is a pre-indexed source, which is built from the community repository. At its core, the `winget` source is just a database file, called `index.db`. When a user runs `winget search`, the CLI runs several queries against the index (index.db) to come up with a list of the packages matching the search criteria. 

My suggestion is, take all of the results that are returned and store them in a temporary database file. Then, a user can use *any* of the default search parameters when refining  the query.